### ISSUE TITLE
array-bracket-newline should allow for line breaks for single element

### ISSUE THREAD

**Issue by mockdeep** [eslint/eslint#9103]
* **ESLint Version:** 4.0.0

It would be nice if the `array-bracket-newline` rule allowed wrapping a single element, at least when it would otherwise be over the line length. When we've got a single value in an array that spans over the line length, we like to wrap the array like so:

```js
// considered bad by the rule defaults:
const myArray = [
  { some: 'long', object: 'that', would: 'go', over: 'the', line: 'length' },
]
```

If there is more than one element in the array, the rule doesn't complain:

```js
// fine by the rule defaults:
const myArray = [
  { some: 'long', object: 'that', would: 'go', over: 'the', line: 'length' },
  { yet: 'another', object: 'that', would: 'go', over: 'the', line: 'length' },
]
```

But when there's only one element it either wants me to put it all on the same line or wrap the contained object across multiple lines:

```js
// rule is cool with this, but breaks line length rule:
const myArray = [{ some: 'long', object: 'that', would: 'go', over: 'the', line: 'length' }]

// or do this to make the rule happy:
const myArray = [
  {
    some: 'long',
    object: 'that',
    would: 'go',
    over: 'the',
    line: 'length',
  },
]
```

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. 

Personally, I'm not sure this is a good idea because it seems like a very personal preference. However, let's see what the rest of the team thinks.

**Comment by platinumazure** [comment]
I see this as another good use case for a "consistent" option which allows line breaks when they're not strictly required, as long as linebreaks are either both present or both absent for a given array.

That said, I don't know if that will integrate well with the existing schema for this rule (as in, I haven't checked).

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
